Sources/MessagingInApp/MessagingInApp.swift

```diff
@@ -25,27 +26,58 @@ public class MessagingInApp: ModuleTopLevelObject<MessagingInAppInstance>, Messa
         Self.shared = MessagingInApp()
     }
 
+    // testing constructor
+    internal static func initialize(organizationId: String, eventListener: InAppEventListener?, implementation: MessagingInAppInstance, sdkInitializedUtil: SdkInitializedUtil) {
+        Self.shared = MessagingInApp(implementation: implementation, sdkInitializedUtil: sdkInitializedUtil)
+
+        if let eventListener = eventListener {
+            Self.initialize(organizationId: organizationId, eventListener: eventListener)
+        } else {
+            Self.initialize(organizationId: organizationId)
+        }
+    }
+
     // Initialize SDK module
     public static func initialize(organizationId: String) {
         Self.shared.initialize(organizationId: organizationId)
     }
 
+    public static func initialize(organizationId: String, eventListener: InAppEventListener) {
+        Self.shared.initialize(organizationId: organizationId, eventListener: eventListener)
+    }
+
     public func initialize(organizationId: String) {
-        initialize() // enables features such as setting up hooks
+        guard let implementation = implementation else {
+            sdkNotInitializedAlert("CustomerIO class has not yet been initialized. Request to initialize the in-app module has been ignored.")
+            return
```

Comment:
This is a new idea I am proposing in this PR. I prefer to introduce new features in separate PRs so I can pull this out and make a separate PR if that's helpful. 

When the SDK is not initialized yet, we should at a minimum tell the customer in a log message that they might have configured the CIO SDK wrong. 

If we like this idea, we could implement this idea in the other classes such as push and `CustomerIO` classes. We check if `implementation` class is null and if so, log a message to customer that the SDK has not yet been initialized so the request is ignored.